internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -254,6 +353,293 @@ var _ = Describe("cluster reconcile", func() {
 		mockCtrl.Finish()
 	})
 
+	Context("Mirror Registry", func() {
+		createAgentClusterInstallForMirrorRegistry := func() *hiveext.AgentClusterInstall {
+			cluster := newClusterDeployment(clusterName, testNamespace, defaultClusterSpec)
+			Expect(c.Create(ctx, cluster)).ShouldNot(HaveOccurred())
+			defaultAgentClusterInstallSpec.MirrorRegistryRef = &hiveext.MirrorRegistryConfigMapReference{Name: providedMirrorRegistryCMName, Namespace: testNamespace}
+
+			aci := newAgentClusterInstall(agentClusterInstallName, testNamespace, defaultAgentClusterInstallSpec, cluster)
+			Expect(c.Create(ctx, aci)).ShouldNot(HaveOccurred())
+
+			return getTestClusterInstall()
+		}
+
+		createUserMirrorRegistryConfigmap := func(registryToml string, certificate string) {
+			registryCM := newUserProvidedRegistryCM(registryToml, certificate)
+			Expect(c.Create(ctx, registryCM)).ShouldNot(HaveOccurred())
+		}
+
+		When("the user-provided valid image registry contains correct data", func() {
```

Comment:
Nit, but single example contexts with no setup code can just inline the context in the test name. I think it will look better in failed test output.